process/block/sovereignChainBlock_test.go

```diff
@@ -883,3 +893,269 @@ func TestSovereignShardProcessor_CreateBlock(t *testing.T) {
 		require.Equal(t, expectedBusyIdleSequencePerCall, busyIdleCalled)
 	})
 }
+
+func TestSovereignShardProcessor_ProcessBlock(t *testing.T) {
+	t.Parallel()
+
+	t.Run("nil header handler should error", func(t *testing.T) {
+		sovArgs := createSovChainBlockProcessorArgs()
+		scbp, err := blproc.NewSovereignChainBlockProcessor(sovArgs)
+		require.Nil(t, err)
+
+		_, _, err = scbp.ProcessBlock(nil, &block.Body{}, haveTime)
```

Comment:
Can we please check everywhere when we return error !=nil  that other fields(header, body) are also nil, please?